test/resources/scheme/PureScheme.Test.Recursion.Mutual/lib.sls

```diff
@@ -1,32 +1,32 @@
 (library
   (PureScheme.Test.Recursion.Mutual lib)
   (export foo bar)
-  (import (rnrs))
+  (import (prefix (rnrs) scm:))
 
 
-  (define
+  (scm:define
     foo
-    (lambda
+    (scm:lambda
       (v)
-      (lambda
+      (scm:lambda
         (v1)
-        (cond
-          ((= v 0) ((foo 1) v1))
-          ((= v 1) ((bar 2) v1))
-          ((= v 2) v1)
-          (#t 0)
-          (else (error #f "Failed pattern match"))))))
+        (scm:cond
+          ((scm:and (scm:= v 0) #t) ((foo 1) v1))
+          ((scm:and (scm:= v 1) #t) ((bar 2) v1))
+          ((scm:and (scm:= v 2) #t) v1)
+          ((scm:and #t #t) 0)
+          (scm:else (scm:error #f "Failed pattern match"))))))
 
-  (define
+  (scm:define
     bar
-    (lambda
+    (scm:lambda
       (v)
-      (lambda
+      (scm:lambda
         (v1)
-        (cond
-          ((= v 0) ((bar 1) v1))
-          ((= v 1) ((foo 2) v1))
-          ((= v 2) v1)
-          (#t 0)
-          (else (error #f "Failed pattern match"))))))
+        (scm:cond
+          ((scm:and (scm:= v 0) #t) ((bar 1) v1))
+          ((scm:and (scm:= v 1) #t) ((foo 2) v1))
+          ((scm:and (scm:= v 2) #t) v1)
+          ((scm:and #t #t) 0)
+          (scm:else (scm:error #f "Failed pattern match"))))))
```

Comment:
The implementation of `cond` changed?

Reply:
No, what's happening here is that we're missing the simplification optimization for `and`.
We pattern match on function application of `and`, but now we have to match on `scm:and`.

Reply:
I see. I think we should fix this here so that the generated code doesn't change this much